alpakaConfig.cmake

```diff
@@ -885,10 +885,10 @@ ELSE()
 
     # GNU
     IF(CMAKE_COMPILER_IS_GNUCXX)
-        LIST(APPEND _ALPAKA_COMPILE_OPTIONS_PUBLIC "-ftemplate-depth-512")
+        LIST(APPEND _ALPAKA_COMPILE_OPTIONS_PUBLIC "-ftemplate-depth-1024")
```

Comment:
We can (and should) remove this for GNU if we are not supporting GCC<4.8 (which we probably do not).